src/b-splines/indexing.jl

```diff
@@ -24,14 +24,21 @@ function hessian_coefficients(::Type{IT}, N, dim1, dim2) where IT<:DimSpec{BSpli
     Expr(:block, exs...)
 end
 
-index_gen(::Type{IT}, N::Integer, offsets...) where {IT} = index_gen(iextract(IT, min(length(offsets)+1, N)), IT, N, offsets...)
+function index_gen(::Type{IT}, N::Integer, offsets...) where {IT}
+    idx = index_gen(iextract(IT, min(length(offsets)+1, N)), IT, N, offsets...)
+    if v"0.7-" ≤ VERSION < v"0.7.0-beta2.119"
```

Comment:
Let's add `@static` to this condition. 

Although this is a meta-programming function and having an explicit branch shouldn't impact performance of evaluating an interpolation, removing the branch at parse time makes sense to me.